backend/src/main/resources/db/migration/V1.10__pipeline_version_per_organism.sql

```diff
@@ -0,0 +1,116 @@
+create view external_metadata_view as
+select
+    cpd.accession,
+    cpd.version,
+    all_external_metadata.updated_metadata_at,
+    -- Combines metadata from preprocessed data with any external metadata updates
+    -- If there's no external metadata, just use the preprocessed data's metadata
+    -- If there is external metadata, merge it with preprocessed data (external takes precedence)
+    case
+        when all_external_metadata.external_metadata is null then
+            jsonb_build_object('metadata', (cpd.processed_data->'metadata'))
+        else
+            jsonb_build_object(
+                'metadata',
+                (cpd.processed_data->'metadata') || all_external_metadata.external_metadata
+            )
+    end as joint_metadata
+from
+    (
+        -- Get only the preprocessed data for the current pipeline version
+        select sepd.*
+        from
+            sequence_entries_preprocessed_data sepd
+            join sequence_entries se
+                on sepd.accession = se.accession and sepd.version = se.version
+            join current_processing_pipeline cpp
+                on se.organism = cpp.organism and sepd.pipeline_version = cpp.version
+    ) cpd
+    left join all_external_metadata on
+        all_external_metadata.accession = cpd.accession
+        and all_external_metadata.version = cpd.version;
+
+drop view if exists sequence_entries_view;
+
+create view sequence_entries_view as
+select
+    se.*,
+    sepd.started_processing_at,
+    sepd.finished_processing_at,
+    sepd.processed_data as processed_data,
+    sepd.processed_data || em.joint_metadata as joint_metadata,
+    case
+        when se.is_revocation then (select version from current_processing_pipeline
+                                    where organism = se.organism)
+        else sepd.pipeline_version
+    end as pipeline_version,
+    sepd.errors,
+    sepd.warnings,
+    case
+        when se.released_at is not null then 'APPROVED_FOR_RELEASE'
+        when se.is_revocation then 'PROCESSED'
+        when sepd.processing_status = 'IN_PROCESSING' then 'IN_PROCESSING'
+        when sepd.processing_status = 'PROCESSED' then 'PROCESSED'
+        else 'RECEIVED'
+    end as status,
+    case
+        when sepd.processing_status = 'IN_PROCESSING' then null
+        when sepd.errors is not null and jsonb_array_length(sepd.errors) > 0 then 'HAS_ERRORS'
+        when sepd.warnings is not null and jsonb_array_length(sepd.warnings) > 0 then 'HAS_WARNINGS'
+        else 'NO_ISSUES'
+    end as processing_result
+from
+    sequence_entries se
+    left join sequence_entries_preprocessed_data sepd on
+        se.accession = sepd.accession
+        and se.version = sepd.version
+    left join current_processing_pipeline ccp on  -- join or left join?
+        se.organism = ccp.organism
+        and sepd.pipeline_version = ccp.version
+    left join external_metadata_view em on
+        se.accession = em.accession
+        and se.version = em.version;
```

Comment:
The `sequence_entries_view` is indeed wrong, it is now showing not just the data associated to the current preprocessing version but to all version:

![image](https://github.com/user-attachments/assets/75d1f9b4-cabf-40fc-8c7c-6f43cbf4519a)

Reply:
I think that @fhennig's original implementation was correct and that my suggestion [here](https://github.com/loculus-project/loculus/pull/3534#discussion_r1933793489) and [here](https://github.com/loculus-project/loculus/pull/3534#discussion_r1933802895) to simplify the SQL code broke it.